src/asset_manager/asset_manager.rs

```diff
@@ -280,6 +320,101 @@ impl AssetStore for DirectoryStore {
     }
 }
 
+impl AssetLoaderRaw for imagefmt::Image<u8> {
+    fn from_raw(_: &Assets, data: &[u8]) -> Option<imagefmt::Image<u8>> {
+        imagefmt::read_from(&mut Cursor::new(data), imagefmt::ColFmt::RGBA).ok()
+    }
+}
+
+impl AssetLoader<Texture> for imagefmt::Image<u8> {
+    fn from_data(assets: &mut Assets, image: imagefmt::Image<u8>) -> Option<Texture> {
+        let pixels = image.buf.chunks(4).map(|p| [p[0], p[1], p[2], p[3]]).collect::<Vec<_>>();
+        let chunked = pixels.as_slice().chunks(image.w).collect::<Vec<_>>();
```

Comment:
`raw` in `TextureLoadData` are slices on images/textures. This is not a 2d array of image data!
For data loaded from an image this means a single slice of pixel data.

This should also fix the mipmap issue:
```Rust
let pixels = image.buf.chunks(4).map(|p| [p[0], p[1], p[2], p[3]]).collect::<Vec<_>>();
AssetLoader::from_data(assets, TextureLoadData {
            kind: Kind::D2(image.w as u16, image.h as u16, AaMode::Single),
            raw: &[pixels.as_slice()],
       })